### PR TITLE
Include template pihole-FTL.conf

### LINKED ISSUE #2309: No file "/etc/pihole/pihole-FTL.conf" installed by default
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`1`

---
**Expected behaviour:**

At the completion of a basic installation, the user should have a commented-out /etc/pihole/pihole-FTL.conf file installed

**Actual behaviour:**

The default installation does not install such a file

**Steps to reproduce:**

Ran the master installer from curl, then checked out to Release 4.0 branch.

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

Not applicable

**Troubleshooting undertaken, and/or other relevant information:**

Since this file is required to make any of the changes that are discussed in the FTLDNS section of the V4.0 documentation, it is likely that users may create one.  Having a file installed by default is recommended for two reasons:

1.  This becomes a good starting place for a user to remove comments or change variables, essentially a common template from which they all work, and 

2.  Simplifies support because all the installations start with this file in place - no back and forth about if the file is there, how to make a file, what to put in the file, etc.

> * You must follow the template instructions. Failure to do so will result in your issue being closed.
> * Please [submit any feature requests here](https://discourse.pi-hole.net/c/feature-requests), so it is votable and trackable by the community.
> * Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.
> * Detail helps us understand and resolve an issue quicker, but please ensure it's relevant.
> * _This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### PR DESCRIPTION
**Pull request by dschaper**
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [ ] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**
Fixes #2309 
Include template pihole-FTL.conf for FTLDNS.

### CODE REVIEW

File: advanced/Templates/pihole-FTL.conf [hunk 1]
```diff
@@ -0,0 +1,78 @@
+## Listen only for local socket connections or permit all connections
+## Options: localonly, all
+
+#SOCKET_LISTENING=localonly
```
[the remark this answers is not in the document]
Comment (dschaper) [reply advanced/Templates/pihole-FTL.conf]: Like this?

```
## Allow FTL to analyze AAAA queries from pihole.log?
## Options: yes, no

#AAAA_QUERY_ANALYSIS=yes


## Should FTL try to resolve IPv6 addresses to host names?
## Options: yes, no

#RESOLVE_IPV6=yes


## Description
## Options:

#OPTION
```

[the remark this answers is not in the document]
Comment (Mcat12) [reply advanced/Templates/pihole-FTL.conf]: Ya, but it is a lot of white space. The [dnsmasq config template](https://github.com/pi-hole/pi-hole/blob/master/advanced/dnsmasq.conf.original) does the no space between description and option thing, which seems standard.

It confuses me a bit with just one space between everything, but if it's an issue changing then that is fine.

[the remark this answers is not in the document]
Comment (dschaper) [reply advanced/Templates/pihole-FTL.conf]: Here's another couple of looks to the format.

```
## Allow FTL to analyze AAAA queries from pihole.log?
## Options: yes, no
#AAAA_QUERY_ANALYSIS=yes

## Should FTL try to resolve IPv6 addresses to host names?
## Options: yes, no
#RESOLVE_IPV6=yes

## Description
## Options:
#OPTION
```

```
## AAAA Query Analysis
## Allow FTL to analyze AAAA queries from pihole.log?
## Options: yes, no
#AAAA_QUERY_ANALYSIS=yes

## Resolve IPv6
## Should FTL try to resolve IPv6 addresses to host names?
## Options: yes, no
#RESOLVE_IPV6=yes

## Option Name
## Description
## Options:
#OPTION
```

[the remark this answers is not in the document]
Comment (Mcat12) [reply advanced/Templates/pihole-FTL.conf]: The last one looks good, if you also like it?

[the remark this answers is not in the document]
Comment (dschaper) [reply advanced/Templates/pihole-FTL.conf]: I agree, I think the last one is the best of all options. JFB has some additions in the way of a header that he'd like to see added, so I'll push the changes to the format as it is now, WIP it and let him make some comments.

**Pull request merged**